esp-hal-common/src/peripheral.rs

```diff
@@ -353,7 +329,7 @@ mod peripheral_macros {
 
             impl crate::peripheral::sealed::Sealed for $name {}
         };
-        ($(#[$cfg:meta])? $name:ident => false) => {
+        ($(#[$cfg:meta])? $name:ident => $base:ident) => {
```

Comment:
I feel like swapping these parameters is more intuitive (you star with the base and go toward the final type name). I know this has been commented on already, and I won't block merging based on this alone, but this is pretty confusing in its current form, IMO.